go/test/endtoend/transaction/twopc/stress/stress_test.go

```diff
@@ -42,6 +42,109 @@ import (
 	"vitess.io/vitess/go/vt/schema"
 )
 
+var (
+	// idVals are the primary key values to use while creating insert queries that ensures all the three shards get an insert.
+	idVals = [3]int{
+		4, // 4 maps to 0x20 and ends up in the first shard (-40)
+		6, // 6 maps to 0x60 and ends up in the second shard (40-80)
+		9, // 9 maps to 0x90 and ends up in the third shard (80-)
+	}
+)
+
+func TestSettings(t *testing.T) {
+	testcases := []struct {
+		name            string
+		commitDelayTime string
+		queries         []string
+		verifyFunc      func(t *testing.T, vtParams *mysql.ConnParams)
+	}{
+		{
+			name:            "No settings changes",
+			commitDelayTime: "5",
+			queries:         append([]string{"begin"}, getMultiShardInsertQueries()...),
+			verifyFunc: func(t *testing.T, vtParams *mysql.ConnParams) {
+				// There is nothing to verify.
+			},
+		},
+		{
+			name:            "Settings changes before begin",
+			commitDelayTime: "5",
+			queries: append(
+				append([]string{`set @@time_zone="+10:30"`, "begin"}, getMultiShardInsertQueries()...),
```

Comment:
nit: multiple append might not be needed.

Reply:
this is done multiple places

Reply:
They're required. getMultiShardInsertQueries is returning a list of strings, so they need to be passed with ellipses. That also means that anything after it, requires a second append call.

Reply:
I missed there were more statements after that.